### PR TITLE
Avoid cacheing restricted pages

### LINKED ISSUE #107: Remedy lack of `no-cache` headers for passworded pages
As noted in https://github.com/wagtail/wagtail/issues/5072 amongst other places, Wagtail is unopinionated about cache-control headers, including on pages that are private/password protected. This can result in restricted content being caught in a CDN or other cache.

We should tweak things, either with a third-party package or code customisation, to avoid this situation + add regression tests to confirm all is OK

### PR DESCRIPTION
**Pull request by stevejalim**
By default, Wagtail is unopinionated about cache-control headers, so we need to be sure that pages with restrictions are not cached anywhere in the chain. We do this by setting the same headers as  Django's `never_cache` helper for pages that have had view restrictions applied to them

Resolves #107

### CODE REVIEW

File: birdbox/.env-example [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
 DISABLE_SSL=True
+RATELIMIT_ENABLE=False
 
-# Change to True if you want to use SSO locally, else you'll use username+password auth
-USE_SSO_AUTH=False
+EMAIL_BACKEND="django.core.mail.backends.console.EmailBackend"
+
+# If you want to use Google Cloud Storage locally, you'll need to set these
+# variables. See https://django-storages.readthedocs.io/en/latest/backends/gcloud.html
+GS_BUCKET_NAME="gcs-bucket-name-here"
```
Comment (pmac) [inline birdbox/.env-example:9]: What are we using GCS buckets for? And shouldn't these be commented out if this is meant to be coppied for an initial working `.env`?

Comment (pmac) [reply birdbox/.env-example:9]: Uploading images for pages I assume? I just don't see anything else in this PR dealing with that.

Comment (stevejalim) [reply birdbox/.env-example:9]: GCS buckets are where we store editor-uploaded image for Wagtail.

And yeap, this is a template for a local .env file (with the assumption that one might want to test/work on something with cloud storage)

Comment (stevejalim) [reply birdbox/.env-example:9]: And yeah, this was a case of mopping up as I was going - not ideal to bundle into this branch, I admit

**Comment by stevejalim** [comment]
Tested locally and def working:

Screenshot 2023-10-02 at 12 59 33

**Pull request merged**